src/dist/replication/common/duplication_common.cpp

```diff
@@ -43,6 +43,15 @@ namespace replication {
     return it->second;
 }
 
+/*extern*/ const char *duplication_fail_mode_to_string(duplication_fail_mode::type fmode)
+{
+    auto it = _duplication_fail_mode_VALUES_TO_NAMES.find(fmode);
```

Comment:
duplication_fail_mode::type is just a struct enum, you can set any value you like. If you set a invalid value, server will coredump here?

Reply:
Yes, if so, something must be wrong. Fail early is necessary.